### DISCUSSION TITLE
"Warning: Could not identify external compartment by name..." from build()

### DISCUSSION THREAD

**Discussion by GeraldineKong** [micom-dev/micom#119] [Q&A] [answered]
Hello! I'm just starting to use Micom and finding it incredibly useful!
I have just a small question regarding a warning that showed up while trying to model the gut microbiome community with the code: `build(tax, "carveme_refseq84_species.qza", "models", solver="gurobi", cutoff=1e-4, threads=10)`

```
WARNING  Could not identify an external          boundary_types.py:90
                    compartment by name and choosing one                        
                    with the most boundary reactions. That                      
                    might be complete nonsense or change                        
                    suddenly. Consider renaming your                            
                    compartments using `Model.compartments`                     
                    to fix this.            
```

What does the warning mean and is it something that I have to keep in mind of while interpreting the downstream results? Also, is there some sort of quality measure that indicates how good the community model is? For example maybe I have set the abundance threshold too high and the model is not representative of the real sample....

**Comment by cdiener** [comment, marked as answer]
Sorry for the late reply. In this instance there is nothing to worry about. For an external compartment cobrapy requires it to (1) contain the majority of transport reactions and (2) have a recognized id/name. Otherwise it will still go with the most likely one but throw a warning. CARVEME adds a `C_` prefix to compartment names like `C_e` so point cobrapy does not recognize the name, but it is still the external compartment so everything will work as expected.

**Comment by TKuehnast** [comment]
Good to know. Using Qiime2 2022.5 with Micom 0.30.5. there was no error, now in Qiime2 2023.5 with Micom 0.32.5 on the same dataset this error appeared and I already wondered how I could possibly get this debugged and what in the world I did wrong again.
